cql3/statements/select_statement.cc

```diff
@@ -1982,6 +1982,12 @@ std::unique_ptr<prepared_statement> select_statement::prepare(data_dictionary::d
         }
     }
 
+    for (auto& ps : prepared_selectors) {
+        if (expr::contains_bind_marker(ps.expr)) {
+            throw exceptions::invalid_request_exception("Unsupported bind marker in SELECT clause selector");
+        }
```

Comment:
Why? It should work as long as we pass query_option during evaluation.

It's useful as SELECT my_map[:variable_field] FROM tab.

Reply:
I need to look more into this to get it to work, but I think it shouldn't be too hard. Unfortunately I'm not very familiar with this area.

Reply:
See selection.cc, it calls expr::evaluate() with `inputs` that contain a reference to query_options. That's enough for it to work.

Reply:
(the change to processes_selection ensures it chooses selectors_with_processing over the alternative that doesn't call evaluate)

Reply:
I added support of `SELECT m[?]` in a new commit

Reply:
I think this will be easier to review if you merge the two commits, instead of the first commit printing an error and the second commit changing it to the implementation.

Reply:
done